DataDog/ecommerce-workshop
# Allow dynamic route mapping for discounts and ads

#13 

This allows frontend to connect to the `ads` and `discounts` on a configurable path set at runtime via the `deploy/docker-compose/.env` file:

The values default to the docker routes `http://advertisements` and `http://discounts`

```
ADS_ROUTE="http://advertisements"
DISCOUNTS_ROUTE="http://discounts"
```

I've also updated the developer documentation to accurately reflect how to make a core change across all frontend versions.

**How to test**
1. Pull down this branch locally, [rebuild frontend code](https://github.com/DataDog/ecommerce-workshop/blob/main/development.md#recreating-the-code) , change the ports in `deploy/docker-compose/.env`, and start the app via the `baseline` compose file e.g. 

```POSTGRES_USER=postgres POSTGRES_PASSWORD=postgres DD_API_KEY=[API KEY] make local-baseline-start```

2. Visit http://localhost and verify that the homepage loads both discounts and ads
